privacy_partner_report/controllers/main.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+import base64
+import json
+import time
+import zlib
+from openerp import http
+from openerp.http import request, content_disposition
+from openerp.service.report import exp_report, exp_report_get
+
+from openerp.addons.web.controllers.main import serialize_exception
+from openerp.addons.web.controllers.main import Reports
+
+
+class ReportsExtended(Reports):
+
+    def _update_report_datas_or_data(self, action, report_ids, report_data):
+        if 'datas' in action:
+            if 'ids' in action['datas']:
+                report_ids = action['datas'].pop('ids')
+            report_data.update(action['datas'])
+        elif 'data' in action:
+            if 'ids' in action['data']:
+                report_ids = action['data'].pop('ids')
+            report_data.update(action['data'])
+        return report_ids, report_data
+
+    @http.route('/web/report', type='http', auth="user")
+    @serialize_exception
+    def index(self, action, token):
```

Comment:
I'm not sure if I'm getting this right, but... is this method here just as a workaround to https://github.com/odoo/odoo/pull/24964?

If so, I think the workaround could be much simpler, similar to:

```python
def index(self, action, token):
    try:
        action["datas"] = action["data"]
    except KeyError:
        pass
    return super(ReportsExtended, self).index(action, token)
```

Would that be enough?

Also please remember to tag hacky code with `# HACK https://github.com/odoo/odoo/pull/24964`

Reply:
No, that simpler workaround wouldn't work. I will tag the hack if the PR resolves negatively. I just uploaded the hack temporarily, just because this PR could be tested fine in Runbot.

Reply:
I'm pretty sure you're right, just could you explain a little bit why wouldn't it work?

Reply:
Ok. I will give a try 🤔

Reply:
```python
Odoo Server Error

Traceback (most recent call last):
  File "/.../OCB/addons/web/controllers/main.py", line 71, in wrap
    return f(*args, **kwargs)
  File "/.../data-protection/privacy_partner_report/controllers/main.py", line 33, in index
    action["datas"] = action["data"]
TypeError: string indices must be integers
```

Reply:
action is just a string when entering Reports controller

Reply:
And if you prepend this?:

```python
action = json.loads(action)
```

You'd have to convert it back to JSON before calling `super()` also.

Reply:
Yeah, that works. Thanks.